src/commands/apps-list.ts

```diff
@@ -0,0 +1,60 @@
+import * as clc from "cli-color";
+import * as ora from "ora";
+// TODO(caot): Replace with proper import
+import Table = require("cli-table");
+
+import * as Command from "../command";
+import * as getProjectId from "../getProjectId";
+import { listFirebaseApps } from "../management/list";
+import { AppMetadata, AppPlatform } from "../management/metadata";
+import * as FirebaseError from "../error";
+import * as requireAuth from "../requireAuth";
+import * as logger from "../logger";
+
+const NOT_SPECIFIED = clc.yellow("[Not specified]");
+
+function logAppsList(apps: AppMetadata[]): void {
+  if (apps.length > 0) {
+    const tableHead = ["App Display Name", "App ID", "Platform"];
+    const table = new Table({ head: tableHead, style: { head: ["green"] } });
+    apps.forEach(({ appId, displayName, platform }) => {
+      table.push([displayName || NOT_SPECIFIED, appId, platform]);
+    });
+
+    logger.info(table.toString());
+  } else {
+    logger.info(clc.bold("No apps found."));
+  }
+}
+
+module.exports = new Command("apps:list [platform]")
+  .description(
+    "list the registered apps of a Firebase project\n\n" +
+      "Arguments:\n" +
+      "[platform] IOS, ANDROID or WEB (case insensitive)"
+  )
+  .before(requireAuth)
+  .action(
+    async (platform: AppPlatform | string = "", options: any): Promise<AppMetadata[]> => {
+      const projectId = getProjectId(options);
+
+      if (platform && !(AppPlatform as any)[platform.toUpperCase()]) {
```

Comment:
This `AppPlatform as any` object access seems dangerous. The [typescript guide](https://www.typescriptlang.org/docs/handbook/enums.html#reverse-mappings) calls out string enums as not having the reverse mapping created, so I'm not actually sure if this even works.

If the advice above (about having a helper function for `string` -> `AppPlatform` is taken, then this check becomes `platform == AppPlatform.PLATFORM_UNSPECIFIED`, which is much, much easier to reason about.

Reply:
A helper function is definitely a clean approach here. I admit that this is probably my habit to apply mapping everywhere applicable, where a simple switch/case helper function can do the work. FYI I tested the above and it works, but yeah looking at it again it's pretty hacky.
Thanks for pointing this out, changing the code...

Reply:
Done